Accelerator profiles and Habana integration - phase one - draft

chtyler opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Initial draft on accelerator profiles and Habana integration with RHODS for 1.34. 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

jstourac commented on modules/options-for-notebook-server-environments.adoc:
```diff
@@ -157,13 +157,46 @@ a| * Python 3.9
 
+.2+| HabanaAI
+| 2023.2 (Recommended)
+a| * Python 3.9
```
```suggestion
a| * Python 3.8
```

jstourac commented on modules/options-for-notebook-server-environments.adoc:
```diff
@@ -157,13 +157,46 @@ a| * Python 3.9
+* SciPy 1.10
+* Kfp-tekton 1.5
+* PyTorch 2.0
+* Elyra 3.15
```
Would be also nice to have all these sorted alphabetically. But that's an optional thing and probably should be done in a separate commit. I'll try to propose this in upstream of the project too.

chtyler replied:
Sure Jan, because of time constraints, we can do this in a forthcoming release. I'd to create a separate Jira for this work, just to keep it separate from this.

FedeAlonso commented on modules/deleting-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,34 @@
++
+A details page for the custom resource definition (CRD) opens.
+. Click the *Instances* tab.
+. Click the action menu (*&#8942;*) beside the accelerator profile that you want to delete and click *Delete*.
```
The button says 'Delete AcceleratorProfile'

chtyler replied:
Fixed.

FedeAlonso commented on modules/enabling-habana-gaudi-devices.adoc:
```diff
@@ -0,0 +1,56 @@
+.. Enter or paste the YAML code directly in the embedded editor. For example:
++
+[source,yaml]
+---
```
Maybe we can point to this example when creating any accelerator

chtyler replied:
Added this example to the "Creating an accelerator profile" module

FedeAlonso commented on modules/working-with-accelerator-profiles.adoc:
```diff
@@ -0,0 +1,53 @@
+To configure accelerators for your data scientists to use in {productname-short}, you must create an associated accelerator profile. An accelerator profile is a custom resource definition (CRD) on OpenShift that has an AcceleratorProfile resource, and defines the specification of the accelerator.  
+
+For accelerators that are new to your deployment, you must manually configure an accelerator profile for the accelerator. If you use NVIDIA GPUs, an accelerator profile generates automatically after you upgrade to the latest version of {productname-short}, and resides in the `Instances` section of the `acceleratorprofiles.dashboard.opendatahub.io` custom resource definition (CRD). 
+
```
Here I miss everything related with upgrading the operator:
If the user had already GPUs working in the  {productname-short} older version: The AcceleratorPRofile CRD instances will be created automatically.
If the user add GPUs after upgrading or after an operator fresh install of the new version: The acceleratorProfile CRD instances should be created manually

FedeAlonso replied:
@chtyler This one is still unfinished?

chtyler replied:
Now fixed.

chtyler commented on modules/creating-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,35 @@
+
+.Procedure
+. In the {openshift-platform} web console, in the *Administrator* perspective, click *Administration* -> *CustomResourceDefinitions*.
+. Click the `acceleratorprofiles.dashboard.opendatahub.io` custom resource definition (CRD).
```
Can you recommend another way of doing this?

chtyler replied:
I think it's easier just to use the search bar on the CustomResourceDefinitions view. Enter AcceleratorProfile into the search bar with "Name" selected on the search bar, and it should appear. @FedeAlonso

chtyler replied:
I have changed all of the procedures to make it clearer.

chtyler commented on modules/deleting-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,34 @@
++
+The *Delete accelerator profile* dialog opens.
+. Enter the name of the accelerator profile in the text field to confirm that you intend to delete it.
+. Click *Delete accelerator profile*.
```
Fixed.

chtyler commented on modules/deleting-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+. Click the action menu (*&#8942;*) beside the accelerator profile that you want to delete and click *Delete*.
++
+The *Delete accelerator profile* dialog opens.
+. Enter the name of the accelerator profile in the text field to confirm that you intend to delete it.
```
I removed this step. Fixed.

chtyler commented on modules/creating-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,35 @@
+. Click *Create AcceleratorProfile*. 
++ 
+The *Create AcceleratorProfile* page opens and displays an embedded YAML editor.
+. Enter or paste the YAML code directly in the embedded editor.
```
Done.

grainnejenningsRH commented on modules/options-for-notebook-server-environments.adoc:
```diff
@@ -157,13 +157,46 @@ a| * Python 3.9
 *Container size* controls the number of CPUs, the amount of memory, and the minimum and maximum request capacity of the container.
 +
-*Number of GPUs* specifies the number of graphics processing units attached to the container.
+*Accelerators* Specifies the accelerators available on your notebook server.
```
Looks like the 's' at the start of 'Specifies' needs to be lowercase here.

chtyler replied:
Fixed.

grainnejenningsRH commented on modules/overview-of-accelerators.adoc:
```diff
@@ -0,0 +1,28 @@
+= Overview of accelerators
+
+[role='_abstract']
+If you work with large data sets, you can use accelerators to optimize the performance of your data science models in {productname-short}. Accelerators enable you to scale your work, reduce latency, and increase productivity. You can use accelerators in {productname-short} to assist your data scientists in the following tasks:
```
Could avoid 'enable you' here by adjusting slightly to something like 'With accelerators, you can scale your work...'

grainnejenningsRH commented on modules/creating-a-project-workbench.adoc:
```diff
@@ -4,15 +4,15 @@
 
 [role='_abstract']
-To examine and work with data models in an isolated area, you can create a workbench. This workbench enables you to create a new Jupyter notebook from an existing notebook container image to access its resources and properties. For data science projects that require data to be retained, you can add container storage to the workbench you are creating.
+To examine and work with data models in an isolated area, you can create a workbench. This workbench enables you to create a Jupyter notebook from an existing notebook container image to access its resources and properties. For data science projects that require data retention, you can add container storage to the workbench you are creating. If you require extra power for use with large datasets, ...
```
We could avoid 'enables you' here by adjusting slightly to something like, 'You can use this workbench to...'

chtyler replied:
Changed.

grainnejenningsRH commented on modules/creating-a-project-workbench.adoc:
```diff
@@ -23,14 +23,18 @@ The *Data science projects* page opens.
+.. Optional: In the *Description* field, enter a description to define your workbench.
+.. In the *Notebook image* section, complete the relevant fields to specify the most suitable notebook image to use with your workbench. 
+... From the *Image selection* list, select a notebook image from the list.
+.. In the *Deployment size* section, complete the relevant fields to specify the size of your deployment instance.
```
Same comment as above RE: 'complete the relevant fields'. 

Could potentially cut down to:
'In the *Deployment size* section, specify the size of your deployment instance.'

chtyler replied:
Changed.

grainnejenningsRH commented on modules/creating-a-project-workbench.adoc:
```diff
@@ -23,14 +23,18 @@ The *Data science projects* page opens.
+.. In the *Notebook image* section, complete the relevant fields to specify the most suitable notebook image to use with your workbench. 
+... From the *Image selection* list, select a notebook image from the list.
+.. In the *Deployment size* section, complete the relevant fields to specify the size of your deployment instance.
+... From the *Container size* list, select a suitable container size for your server.
```
Same comment as above RE: 'suitable'.

grainnejenningsRH commented on modules/creating-a-project-workbench.adoc:
```diff
@@ -23,14 +23,18 @@ The *Data science projects* page opens.
+... From the *Image selection* list, select a notebook image from the list.
+.. In the *Deployment size* section, complete the relevant fields to specify the size of your deployment instance.
+... From the *Container size* list, select a suitable container size for your server.
+... Optional: From the *Accelerator* field, select an accelerator from the list. 
```
Same comment as above RE: list fields.

'From the *Accelerator* list, select an accelerator.'

chtyler replied:
Changed.

grainnejenningsRH commented on modules/creating-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,35 @@
+. Click *Create*.
+
+.Verification
+* The *Accelerator* list displays on the *Start a notebook server* page. After you select an accelerator, the *Number of accelerators* field appears allowing you to choose the number of accelerators for your notebook server.  
```
- Same as above RE: 'displays' > 'appears'.
- Could avoid use of 'allowing you' by rephrasing slightly to something like:
'After you select an accelerator, the *Number of accelerators* field appears, which you can use to choose the number of accelerators for your notebook server.'

chtyler replied:
Changed.

grainnejenningsRH commented on modules/deleting-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+.Prerequisites
+* You have logged in to {openshift-platform}.
+* You have the `cluster-admin` role in {openshift-platform}.
+* The `acceleratorprofiles.dashboard.opendatahub.io` custom resource definition (CRD) exists in your deployment and contains an accelerator profile.
```
You've updated 'contained' to 'has' elsewhere in this PR, if that's a 'plain language' swap we want to make consistently everywhere.

grainnejenningsRH commented on modules/deleting-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+. Click *Delete accelerator profile*.
+
+.Verification
+* The accelerator profile no longer displays in the *Accelerator* list on the *Start a notebook server* page.   
```
For this and line 27, same comment as above RE: 'displays' > 'appears'.

grainnejenningsRH commented on modules/enabling-habana-gaudi-devices.adoc:
```diff
@@ -0,0 +1,56 @@
+.. Click the *Instances* tab.
+.. Click *Create AcceleratorProfile*. 
++ 
+The *Create AcceleratorProfile* page opens and displays an embedded YAML editor.
```
Same comment as above RE: cutting out 'and displays'.

grainnejenningsRH commented on modules/enabling-habana-gaudi-devices.adoc:
```diff
@@ -0,0 +1,56 @@
+.. Click *Create AcceleratorProfile*. 
++ 
+The *Create AcceleratorProfile* page opens and displays an embedded YAML editor.
+.. Enter or paste the YAML code directly in the embedded editor. For example:
```
Same question as above RE: YAML code coming from here? 

Aware of advising 'to paste the YAML code' if we've not referenced copying specific code elsewhere.

grainnejenningsRH commented on modules/enabling-habana-gaudi-devices.adoc:
```diff
@@ -0,0 +1,56 @@
+.. Click *Create*.
+
+.Verification
+* From the *Administrator* perspective, the following Operators display on the *Operators* -> *Installed Operators* page.
```
Same comment as above RE: 'display' > 'appear'.

grainnejenningsRH commented on modules/launching-jupyter-and-starting-a-notebook-server.adoc:
```diff
@@ -32,15 +32,16 @@ If you credentials are accepted, the *Notebook server control panel* opens displ
-.. Select the *Container size* for your server.
-.. Optional: Select the *Number of GPUs* (graphics processing units) for your server.
+.. From the *Container size* list, select a suitable container size for your server.
+.. Optional: From the *Accelerator* field, select an accelerator from the list. 
```
Same as comment above RE: list elements & repetition.

Could cut down to:
'From the *Accelerator* list, select an accelerator.'

grainnejenningsRH commented on modules/launching-jupyter-and-starting-a-notebook-server.adoc:
```diff
@@ -32,15 +32,16 @@ If you credentials are accepted, the *Notebook server control panel* opens displ
-.. Optional: Select the *Number of GPUs* (graphics processing units) for your server.
+.. From the *Container size* list, select a suitable container size for your server.
+.. Optional: From the *Accelerator* field, select an accelerator from the list. 
+.. Optional: If you selected an accelerator in the preceding step, from the *Number of accelerators* list, select the number of accelerators to use.  
```
Could reduce repetition in this line by adjusting to:

'Optional: If you selected an accelerator in the preceding step, from the *Number of accelerators* list, select a value.'

chtyler replied:
Changed.

grainnejenningsRH commented on modules/adding-a-model-server-to-your-data-science-project.adoc:
```diff
@@ -47,23 +47,23 @@ The *Add model server* dialog opens.
 ====
-
+. Optional: From the *Accelerator* field, select an accelerator from the list. 
+. Optional: If you selected an accelerator in the preceding step, from the *Number of accelerators* list, select the number of accelerators to use.  
```
Could trim this down to
'Optional: If you selected an accelerator in the preceding step, from the Number of accelerators list, select a value.'

chtyler replied:
Changed.

grainnejenningsRH commented on modules/creating-a-project-workbench.adoc:
```diff
@@ -23,14 +23,18 @@ The *Data science projects* page opens.
+.. In the *Deployment size* section, complete the relevant fields to specify the size of your deployment instance.
+... From the *Container size* list, select a suitable container size for your server.
+... Optional: From the *Accelerator* field, select an accelerator from the list. 
+... Optional: If you selected an accelerator in the preceding step, from the *Number of accelerators* list, select the number of accelerators to use.  
```
Could trim this down to
'Optional: If you selected an accelerator in the preceding step, from the Number of accelerators list, select a value.'

grainnejenningsRH commented on modules/configuring-a-recommended-accelerator-tag.adoc:
```diff
@@ -0,0 +1,38 @@
+A details page opens.
+. Click the *YAML* tab.
++ 
+A page with an embedded YAML editor opens.
```
Where is the YAML code coming from here? Is it a specific chunk of code, or are we just trying to say 'you need to enter YAML code in this editor'?

chtyler replied:
If they've manually written some YAML code elsewhere, such as in a text editor, they can copy the YAML from the text editor and then paste it in to the embedded YAML editor. If they don't have any pre-prepared YAML, they will need to enter the code manually in the embedded editor.

grainnejenningsRH commented on modules/configuring-a-recommended-accelerator-tag.adoc:
```diff
@@ -0,0 +1,38 @@
+. Click *Save*.
+
+.Verification
+* When your data scientists select an accelerator with a specific notebook image, a recommended tag displays next to the corresponding accelerator. 
```
'displays' > 'appears' for this line and the next.

chtyler replied:
Changed.

chtyler commented on modules/options-for-notebook-server-environments.adoc:
```diff
@@ -6,7 +6,7 @@
 When you start Jupyter for the first time, or after stopping your notebook server, you must select server options in the *Start a notebook server* wizard so that the software and variables that you expect are available on your server. This section explains the options available in the *Start a notebook server* wizard in detail.
 
-The *Start a notebook server* page is divided into several sections:
+The *Start a notebook server* page comprises of the following sections:
```
Changed.

chtyler commented on modules/updating-a-model-server.adoc:
```diff
@@ -43,23 +43,24 @@ NOTE: You cannot change the *Serving runtime* selection for a model server that
+If you use a _custom_ model-serving runtime with your model server, you must ensure that your custom runtime supports GPUs and is appropriately configured to use them.
 ====
 
+.. Optional: From the *Accelerator* field, select an accelerator from the list. 
```
Changed.

chtyler commented on modules/adding-a-model-server-to-your-data-science-project.adoc:
```diff
@@ -47,23 +47,23 @@ The *Add model server* dialog opens.
 If you are using a _custom_ model-serving runtime with your model server, you must ensure that your custom runtime supports GPUs and is appropriately configured to use them.
 ====
-
+. Optional: From the *Accelerator* field, select an accelerator from the list. 
```
It's a drop-down list where you select one option. I've changed it to your suggestion.

chtyler commented on modules/updating-a-model-server.adoc:
```diff
@@ -43,23 +43,24 @@ NOTE: You cannot change the *Serving runtime* selection for a model server that
 
 .Verification
-* The model server that you updated is displayed in the *Models and model servers* section on the project details page.
+* The model server that you updated displays in the *Models and model servers* section on the project details page.
```
Changed.

chtyler commented on modules/creating-a-project-workbench.adoc:
```diff
@@ -23,14 +23,18 @@ The *Data science projects* page opens.
+.. In the *Name* field, enter a name for your workbench.
+.. Optional: In the *Description* field, enter a description to define your workbench.
+.. In the *Notebook image* section, complete the relevant fields to specify the most suitable notebook image to use with your workbench. 
+... From the *Image selection* list, select a notebook image from the list.
```
They're both drop-down lists. I've changed them to your suggestion.

chtyler commented on modules/updating-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,33 @@
+.Prerequisites
+* You have logged in to {openshift-platform}.
+* You have the `cluster-admin` role in {openshift-platform}.
+* The `acceleratorprofiles.dashboard.opendatahub.io` custom resource definition (CRD) exists in your deployment and contains an accelerator profile.
```
changed.

chtyler commented on modules/working-with-accelerator-profiles.adoc:
```diff
@@ -0,0 +1,53 @@
+
+For accelerators that are new to your deployment, you must manually configure an accelerator profile for the accelerator. If you use NVIDIA GPUs, an accelerator profile generates automatically after you upgrade to the latest version of {productname-short}, and resides in the `Instances` section of the `acceleratorprofiles.dashboard.opendatahub.io` custom resource definition (CRD). 
+
+To indicate suitable accelerators to your data scientists, you can assign specific accelerator profiles to your custom notebook images. This also enables you to manage the accelerators that appear to your data scientist.
```
Changed.

chtyler commented on modules/updating-access-to-a-data-science-project.adoc:
```diff
@@ -40,13 +40,13 @@ The *Permissions* page for the project opens.
 --
 ifndef::self-managed[]
-If you do not have `cluster-admin` or `dedicated-admin` permissions, the *Name* list is not visible. Instead, an input field is displayed enabling you to configure group permissions.
+If you do not have `cluster-admin` or `dedicated-admin` permissions, the *Name* list is not visible. Instead, an input field displays enabling you to configure group permissions.
```
Changed

chtyler commented on modules/creating-a-project-workbench.adoc:
```diff
@@ -23,14 +23,18 @@ The *Data science projects* page opens.
-.. Select the *container size* for your server.
+.. In the *Name* field, enter a name for your workbench.
+.. Optional: In the *Description* field, enter a description to define your workbench.
+.. In the *Notebook image* section, complete the relevant fields to specify the most suitable notebook image to use with your workbench. 
```
Is there a reason we need to include 'the most suitable' or 'relevant fields' here? The data scientist will likely have an idea about what technologies they require for their data science work. Because of this, they will need to select a notebook image that is most suitable for the type of work that they plan to carry out. Some notebook images will not be suitable for their work.

Are we keeping things broad with 'complete the relevant fields' because there are different fields for the different notebook images? The fields are the same. In this case, the "Notebook Image" section is just the title of a section.

grainnejenningsRH replied:
Ok, understood. In that case, I do think we could consider cutting this down as being a bit fluffy. 

From what I've seen of the docs, we assume the user is going to select what they think is most suitable/relevant for every step. It could cause confusion to only call out relevance/suitability on certain steps (or imply that selecting the most suitable/relevant choice _doesn't_ matter on others).

chtyler replied:
Fair point. I'll try and cut out the fluff.

chtyler commented on modules/updating-a-project-workbench.adoc:
```diff
@@ -27,10 +27,15 @@ The *Details* page for the project opens.
-.. Update *description* for your workbench, if applicable.
-.. Select a new *notebook image* to use for your workbench server, if applicable.
-.. Select a new *container size* for your server, if applicable.
+.. In the *Name* field, update the *name* for your workbench, if applicable.
```
Changed.

chtyler commented on modules/creating-an-accelerator-profile.adoc:
```diff
@@ -0,0 +1,35 @@
+. Click *Create AcceleratorProfile*. 
++ 
+The *Create AcceleratorProfile* page opens and displays an embedded YAML editor.
+. Enter or paste the YAML code directly in the embedded editor.
```
If they've manually written some YAML code elsewhere, such as in a text editor, they can copy the YAML from the text editor and then paste it in to the embedded YAML editor. If they don't have any pre-prepared YAML, they will need to enter the code manually in the embedded editor.

grainnejenningsRH replied:
Gotcha, thank you for explaining. I think using '_the_ YAML code' could suggest that it's a specific chunk of code the user needs to enter here - if there's a lot of variation on what users could be entering, that could cause confusion. 

Similar thinking with 'paste' - could be less clear than we want for users that don't have prepped YAML. 

From a concision POV, do we need to include 'or paste', if it's just a method of entering the code? (Unless it's not clear from the UI that the user is able to paste code in.)

jstourac commented on modules/options-for-notebook-server-environments.adoc:
```diff
@@ -157,13 +157,46 @@ a| * Python 3.9
+a| * Python 3.8
+* Habana 1.10
+* TensorFlow 2.12
+* Boto3 1.17
```
I can see this is marked as done but wasn't updated. Was my suggestion wrong?

chtyler replied:
Your suggestion is not wrong, as far as I know. We are still receiving SME Feedback on Habana integration, delaying the progress of this MR.

jstourac replied:
this isn't resolved - boto3 should be `1.26`

chtyler replied:
Fixed.

chtyler commented:
@FedeAlonso - please see the latest changes from the most recent commit. I have addressed your feedback. If you notice anything else, please let me know. @jstourac -  your suggested changes should now be applied.

jstourac commented on modules/working-with-accelerator-profiles.adoc:
```diff
@@ -0,0 +1,55 @@
+
+For accelerators that are new to your deployment, you must manually configure an accelerator profile for the accelerator. If your deployment contained NVIDIA GPUs before upgrading {productname-short}, an accelerator profile generates automatically after you upgrade to the latest version of {productname-short}, and resides in the `Instances` section of the `AcceleratorProfile` custom resource defi ...
+
+If you add NVIDIA GPUs to your deployment after upgrading {productname-short}, or you add them to your deployment after you install a new version of the {productname-short} Operator, you must manually create an accelerator profile for your NVIDIA GPUs.
```
1/ I think we should not mention NVIDIA GPU specifically in this doc unless we are giving some particular example. We should rather use some generic term like `accelerator`, `hardware accelerator`, `hardware accelerator unit` or similar?

2/ These two upgrade information - it's just a one time thing for the upgrade between two particular releases. We should put this note into some Upgrade Notes - that AcceleratorProfile will be created automatically if your deployment has had the hardware accelerator unit pre-configured before the upgrade already.

In general, here we should only have some message like this:

```
For hardware accelerators that are new to your deployment, you must manually configure an accelerator profile for that particular hardware accelerator. If your deployment contained a hardware accelerator before the upgrade, it will be preserved after the upgrade.
```

WDYT?

chtyler replied:
I have reworded the parapgraph based on your suggestion. I have used the term "accelerators".

chtyler replied:
I will update the downstream RHODS upgrade guides with information such as:

For hardware accelerators that are new to your deployment, you must manually configure an accelerator profile for that particular hardware accelerator. If your deployment contained a hardware accelerator before the upgrade, it will be preserved after the upgrade.

jstourac replied:
Thanks! :slightly_smiling_face:

jstourac commented on modules/overview-of-accelerators.adoc:
```diff
@@ -0,0 +1,29 @@
+** To use compute-heavy workloads in your models, you can enable NVIDIA graphics processing units (GPUs) in {productname-short}. 
+**   To enable GPUs on OpenShift, you must install the NVIDIA GPU Operator.
+* Habana Gaudi devices (HPUs)
+** Habana, an Intel company, provides hardware accelerators intended for deep learning workloads. {productname-short} currently supports Habana 1.10.0 and Habana 1.11.0. 
```
I'm not sure what we mean by `support` in the upstream, but until there is an appropriate Habana Operator for version 1.11, we shouldn't include it here.

chtyler replied:
Reworded the sentence to remove the word "support". Removed references to 1.11.

jstourac commented on modules/habana-gaudi-integration.adoc:
```diff
@@ -0,0 +1,38 @@
+====
+Currently, Habana Gaudi integration is only supported in OpenShift 4.11 and 4.12. 
+
+{productname-short} supports version 1.10 and 1.11 of the Habana Gaudi Operator. For information about the supported configurations for versions 1.10 and 1.11 of the Habana Gaudi Operator, see link:https://docs.habana.ai/en/latest/Support_Matrix/Support_Matrix_v1.10.0.html#support-matrix-1-10-0[Support Matrix v1.10.0] and https://docs.habana.ai/en/latest/Support_Matrix/Support_Matrix_v1.11.0.html ...
```
Well, v1.11 is under question still.

chtyler replied:
Removing all references to 1.11 now.

jstourac reviewed (commented):
Hi Chris, I've put some more comments. Hopefully it's all. Except Habana v1.11. We'll see what happens in next hours.

chtyler commented on modules/enabling-habana-gaudi-devices.adoc:
```diff
@@ -0,0 +1,58 @@
+= Enabling Habana Gaudi devices
+
+[role='_abstract']
+Before you can use Habana Gaudi devices in {productname-short}, you install the necessary dependencies and deploy the HabanaAI Operator in your deployment. 
```
We use "must" rather than "have to". Adding "must". Nice spot :)

chtyler commented on modules/options-for-notebook-server-environments.adoc:
```diff
@@ -157,13 +157,46 @@ a| * Python 3.9
 * SciPy 1.10
 
+.2+| HabanaAI
+| 2023.2 (Recommended)
```
For clarification, @harshad16 has given the go-ahead to remove mentions of Habana 1.11 from the docs. So I will remove Habana 1.11 from this table, and the recommended tag.

chtyler replied:
Removed 2023.2 and recommended tag.

jstourac commented on modules/options-for-notebook-server-environments.adoc:
```diff
@@ -157,13 +157,31 @@ a| * Python 3.9
 * Scikit-learn 1.2
 * SciPy 1.10
 
+.2+| HabanaAI
```
Not sure about the `.2+` here - TrustyAI doesn't have it and it's also just one version image :shrug:

chtyler replied:
It's an AsciiDoc formatting thing.

chtyler replied:
Will try and remove it.

chtyler replied:
Removed.

jstourac commented on modules/options-for-notebook-server-environments.adoc:
```diff
@@ -157,13 +157,31 @@ a| * Python 3.9
 * SciPy 1.10
 
+.2+| HabanaAI
+| 2023.1
```
```suggestion
| 2023.1 (Recommended)
```

jstourac replied:
Heh, I realized now that it doesn't make sense to have the `Recommended` here since it's just one version. But whatever. Keep it here :)

jstourac reviewed (changes requested):
Thank you, some more comments. LGTM otherwise.

jstourac reviewed (approved):
LGTM, thank you :slightly_smiling_face:

chtyler commented:
Merging.
